src/java/test/org/apache/zookeeper/test/ClientBase.java

```diff
@@ -368,22 +368,37 @@ static void verifyThreadTerminated(Thread thread, long millis)
         }
     }
 
+    public static File createEmptyTestDir() throws IOException {
+        return createTmpDir(BASETEST, false);
+    }
 
     public static File createTmpDir() throws IOException {
-        return createTmpDir(BASETEST);
+        return createTmpDir(BASETEST, true);
     }
 
-    static File createTmpDir(File parentDir) throws IOException {
+    static File createTmpDir(File parentDir, boolean createInitFile) throws IOException {
         File tmpFile = File.createTempFile("test", ".junit", parentDir);
         // don't delete tmpFile - this ensures we don't attempt to create
         // a tmpDir with a duplicate name
         File tmpDir = new File(tmpFile + ".dir");
         Assert.assertFalse(tmpDir.exists()); // never true if tmpfile does it's job
         Assert.assertTrue(tmpDir.mkdirs());
 
+        // todo not every tmp directory needs this file
```

Comment:
why is this todo here?

Reply:
Unfortunately this todo comment is in the master branch from where I backported these changes in ClientBase.